File: coin_piles.py
def g(a, b):
    """
    We're looking for nonnegative integer solutions x, y to
    a = 2x + y
    b = x + 2y
    """
    three_x = 2 * a - b
    three_y = 2 * b - a

    return three_x >= 0 and three_y >= 0 and three_x % 3 == 0 and three_y % 3 == 0


def main():
    t = int(input())
    for k in range(t):
        a, b = [int(x) for x in input().split()]
        if g(a, b):
            print("YES")
        else:
            print("NO")


main()

File: CoinPiles.cpp
#include <iostream>

using namespace std;

typedef long long ll;

bool g(ll a, ll b) {
  ll three_x = 2 * a - b;
  ll three_y = 2 * b - a;
  return three_x >= 0 && three_y >= 0 && three_x % 3 == 0 && three_y % 3 == 0;
}

void solve() {
  size_t t;
  cin >> t;

  for (size_t i = 0; i < t; i++) {
    ll a, b;
    cin >> a >> b;
    if (g(a, b))
      cout << "YES\n";
    else
      cout << "NO\n";
  }
}

int main() {
  ios::sync_with_stdio(0);
  cin.tie(0);
  solve();
  return (0);
}